app/javascript/components/concept-map/Concept.tsx

```diff
@@ -10,30 +17,37 @@ export const Concept = ({
   handleEnter,
   handleLeave,
   status,
-  isActive,
-  isDimmed,
-  adjacentConcepts,
+  isInactive,
 }: IConcept & {
-  status: ConceptState
-  isActive: boolean
-  isDimmed: boolean
-  adjacentConcepts: string[]
+  status: ConceptStatus
+  isInactive: boolean
 }) => {
+  const [visibility, setVisibility] = useState<Visibility>('hidden')
+  const conceptRef = useRef(null)
+
+  useEffect(() => {
+    const current = conceptRef.current
+    emitConceptElement(slug, current)
+    addVisibilityListener(setVisibility)
+    return () => {
+      emitConceptElement(slug)
+      removeVisibilityListener(setVisibility)
+    }
+  }, [slug, conceptRef])
+
   // Build the class list
   let classes = ['card']
   classes.push(`card-${status}`)
-  if (isActive) {
-    classes.push('card-active')
+  if (isInactive) {
```

Comment:
I think this should still be `isActive`. Having negatively-named variables is generally a bit of a smell. Even though you're only checking for negation, it's better to have `if(!active)` than `if(inactive)`.

Reply:
Fixed in update